packages/http/src/shared.ts

```diff
@@ -5,31 +5,187 @@ export type TActivityResponse = definitions["v1ActivityResponse"];
 export type TActivityId = TActivity["id"];
 export type TActivityStatus = TActivity["status"];
 export type TActivityType = TActivity["type"];
+export type TSignature = definitions["v1SignRawPayloadResult"];
+export type TSignedTransaction =
+  definitions["v1SignTransactionResult"]["signedTransaction"];
 
 export class TurnkeyActivityError extends Error {
-  activityId: TActivityId | null;
-  activityStatus: TActivityStatus | null;
-  activityType: TActivityType | null;
-  cause: Error | null;
+  activityId: TActivityId | undefined;
```

Comment:
Why the change from `null` to `undefined`?

Reply:
I went into this because of type compatibility, and ended up going with undefined for consistency. (functionally, `null == undefined = true`). Specifically, I feel like `param: <Type X> | undefined` better conveys the idea that the absence of the param is okay

Reply:
Is `activityId: TActivityId | undefined;` equivalent to `activityId?: TActivityId;` then?

Reply:
If that's the case I'm a definite +1 on the change; simpler to reason about!